### PR TITLE
[AAP-75386] Forward task environment: block to manager subprocess to …

### PR DESCRIPTION
**Pull request by rohitthakur2590**
…fix SSL cert verification in containerized installs

## Description

- What is being changed?
spawn_manager_process() in process_manager.py now accepts an optional task_env parameter that is overlaid on os.environ before the subprocess is spawned. If SSL_CERT_FILE is set and REQUESTS_CA_BUNDLE is not, it is automatically mapped so the requests library picks it up. In base_action.py, both the ephemeral (connection: local) and experimental persistent paths now resolve self._task.environment via the Jinja2 templar and pass the result as task_env to the manager subprocess.
- Why is this change needed?
The containerized AAP installer sets SSL_CERT_FILE via the task environment: block to point to a private CA bundle. However, self._task.environment was never forwarded to subprocess.Popen() when spawning the manager process, so the manager subprocess had no knowledge of the custom CA. Additionally, the requests library — which the manager uses for all HTTP calls — does not read SSL_CERT_FILE at all; it reads REQUESTS_CA_BUNDLE. Both gaps together caused SSLCertVerificationError on every ansible.platform task in containerized deployments.
- How does this change address the issue?
Task-level environment variables are now resolved (Jinja2 templates expanded) and passed through the call chain — base_action.py  - > spawn_ephemeral_client() - > spawn_manager_process() — so they are present in the manager subprocess environment before the first HTTP request is made. The automatic SSL_CERT_FILE  - > REQUESTS_CA_BUNDLE mapping bridges the gap between Python's ssl stdlib and the requests library, requiring no changes to existing playbooks. All existing callers are unaffected as task_env defaults to None.

Assisted by:  Claude Code
## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

### Steps to Test
1. 
2. 
3. 

### Required Actions

- [ ] Requires documentation updates
  
- [ ] Requires downstream repository changes
  
- [ ] Requires infrastructure/deployment changes
  
- [ ] Requires coordination with other teams
  
- [ ] Blocked by PR/MR: #XXX

### CODE REVIEW

File: plugins/plugin_utils/manager/process_manager.py [hunk 1]
```diff
@@ -220,8 +226,18 @@ def spawn_manager_process(
...
         env = os.environ.copy()
+        if task_env:
+            env.update({k: str(v) for k, v in task_env.items() if v is not None})
+            logger.debug("Applied %d task-level environment variable(s) to manager subprocess", len(task_env))
+
+        # Map SSL_CERT_FILE → REQUESTS_CA_BUNDLE so the requests library picks up
+        # custom CA bundles set by containerised installers.
+        if env.get("SSL_CERT_FILE") and not env.get("REQUESTS_CA_BUNDLE"):
+            env["REQUESTS_CA_BUNDLE"] = env["SSL_CERT_FILE"]
+            logger.debug("Mapped SSL_CERT_FILE → REQUESTS_CA_BUNDLE: %s", env["REQUESTS_CA_BUNDLE"])
```
Comment (dsavineau) [inline plugins/plugin_utils/manager/process_manager.py]: I don't think this is required

We used `SSL_CERT_FILE` in previous version because that was what the old implementation relied on

If `REQUESTS_CA_BUNDLE` works with the new implementation then we can change the installer code rather than doing hack like this in the collection

Comment (rohitthakur2590) [reply plugins/plugin_utils/manager/process_manager.py]: yes agreed! I have removed this .

**Review by komaldesai13 — APPROVED** [review]

**Comment by rohitthakur2590** [comment]
closing this as these fixes will be tracked under PR#182:  https://github.com/ansible/ansible.platform/pull/182
